### PR TITLE
[Form] Ability to clear form errors

### PR DESCRIPTION
**Pull request by colinodell**
Documenting the new feature proposed in https://github.com/symfony/symfony/pull/27580

### CODE REVIEW

**Review by maxhelias — APPROVED** [review]

File: components/form.rst [hunk 1]
```diff
@@ -744,6 +744,11 @@ method to access the list of errors. It returns a
     // a FormErrorIterator instance representing the form tree structure
     $errors = $form->getErrors(true, false);
 
+Clearing Form Errors
+~~~~~~~~~~~~~~~~~~~~
+
+Any errors can be manually cleared using the :method:`Symfony\\Component\\Form\\ClearableErrorInterface::clearErrors` method. This is useful when you'd like to validate the form without showing validation errors to the user (i.e. during a partial AJAX submission or :doc:`dynamic form modification </form/dynamic_form_modification>`).
```
Comment (yceruto) [inline components/form.rst]: `ClearableErrorsInterface`? `s`

Comment (colinodell) [reply components/form.rst]: Good catch! I renamed the interface but forgot to update the docs.  Fixed and amended!

File: components/form.rst [hunk 2]
```diff
@@ -744,6 +744,11 @@ method to access the list of errors. It returns a
     // a FormErrorIterator instance representing the form tree structure
     $errors = $form->getErrors(true, false);
 
+Clearing Form Errors
+~~~~~~~~~~~~~~~~~~~~
+
+Any errors can be manually cleared using the :method:`Symfony\\Component\\Form\\ClearableErrorsInterface::clearErrors` method. This is useful when you'd like to validate the form without showing validation errors to the user (i.e. during a partial AJAX submission or :doc:`dynamic form modification </form/dynamic_form_modification>`).
```
Comment (HeahDude) [inline components/form.rst]: We should precise that clearing the errors makes the form valid. So `clearErrors()` should be called after testing if the form is valid or at least not rely on its state.

**Review by HeahDude — CHANGES_REQUESTED** [review]

File: components/form.rst (hunk 2 above)
Comment (colinodell) [reply components/form.rst]: I have added a new sentence to clarify that.  Thanks!

**Review by HeahDude — APPROVED** [review]
Thanks!

**Review by vudaltsov — APPROVED** [review]

**Comment by javiereguiluz** [comment]
@colinodell thanks for improving Forms DX with this new feature and thanks for providing the docs too. Everything is perfectly explained!

**Pull request merged**